merlin-server/sql/merlin/tables/anchor_validation_status.sql

```diff
@@ -0,0 +1,341 @@
+  then
+    /*
+      Postgres if's don't "short-circuit" -- all clauses are evaluated. Therefore, this query is placed here so that
+      it only runs iff the outer 'if' is true
+    */
+    with recursive anchors(plan_id, activity_id, anchor_id, anchored_to_start, start_offset, total_offset) as (
+      select _plan_id, _activity_id, _anchor_id, _anchored_to_start, _start_offset, _start_offset
+      union
+      select ad.plan_id, ad.id, ad.anchor_id, ad.anchored_to_start, ad.start_offset, anchors.total_offset + ad.start_offset
+      from activity_directive ad, anchors
+      where anchors.anchor_id is not null                               -- stop at plan
+        and (ad.plan_id, ad.id) = (anchors.plan_id, anchors.anchor_id)
+        and anchors.total_offset < '0'                                  -- or, stop at a nonnegative offset
+    )
+    select anchor_id  -- get the id of the activity that the selected activity is anchored to
+    from anchors a
+    where a.total_offset < '0'
+      and not a.anchored_to_start
+      and a.anchor_id is not null
+    limit 1
+    into end_anchor_id;
+
+    if end_anchor_id is not null then
+      raise notice 'Activity Directive % has a net negative offset relative to an end-time anchor on Activity Directive %.', _activity_id, end_anchor_id;
+
+      insert into anchor_validation_status (activity_id, plan_id, reason_invalid)
+      values (_activity_id, _plan_id, 'Activity Directive ' || _activity_id || ' has a net negative offset relative to an end-time' ||
+                                      ' anchor on Activity Directive ' || end_anchor_id ||'.')
+      on conflict (activity_id, plan_id) do update
+        set reason_invalid = 'Activity Directive ' || excluded.activity_id || ' has a net negative offset relative to an end-time' ||
+                             ' anchor on Activity Directive ' || end_anchor_id ||'.';
+
+      return true;
+    end if;
+  end if;
+  return false;
+end
+$$;
+comment on function is_net_negative_end_offset(_activity_id integer, _plan_id integer) is e''
+  'Returns true if the specified activity has a net negative offset from a non-plan activity end-time anchor. Otherwise, returns false.\n'
+  'If true, writes to anchor_validation_status.';
+
+-- An activity may not have a start time before the plan
+create function is_net_negative_plan_start(_activity_id integer, _plan_id integer)
+  returns boolean
+  security definer
+  language plpgsql as $$
+  declare
+    net_offset interval;
+    _anchor_id integer;
+    _start_offset interval;
+    _anchored_to_start boolean;
+  begin
+    select anchor_id, start_offset, anchored_to_start
+    from activity_directive
+    where (id, plan_id) = (_activity_id, _plan_id)
+    into _anchor_id, _start_offset, _anchored_to_start;
+
+    if(_start_offset < '0'  and _anchored_to_start) then -- only need to check if anchored to start or something with a negative offset
```

Comment:
Suggest whitespace tweak
```suggestion
    if (_start_offset < '0' and _anchored_to_start) then -- only need to check if anchored to start or something with a negative offset
```